src/reflect/scala/reflect/internal/Names.scala

```diff
@@ -554,7 +536,7 @@ trait Names extends api.Names {
         if (n ne null) n
         else {
           val next = typeHashtable(h)
-          val typeName = createCompanionName(next)
+          val typeName = createCompanionName(next).asInstanceOf[TypeName]
```

Comment:
Why do we need the cast?

Reply:
A relic of a previous version of this patch, I'll remove.